### DISCUSSION TITLE
Don't show all headings on the table of contents of each page

### DISCUSSION THREAD

**Discussion by fesja** [outline/outline#3973] [Ideas]
Hi, I find the table of contents on each page particularly useful. However, if there are many headings H1-H4, the table of contents becomes too long and it loses its purpose. My suggestion is to reduce the number of headings used in the table of content. Some options:

1) Don't use H4 headings in the table of content. This would apply for everyone. People would have H1-H3 if they have the heading to be shown on the table of content. Easy to code.
2) Allow on a page level to select which heading levels to show in the table of content. This requires a bit more work.
3) Allow on an account level to select which heading levels to show in the table of content. My impression is that this should be in the page level.

I'd go with 1) to have a first iteration and gather feedback from other users.

regards

**Comment by tommoor** [comment]
I don't feel strongly about this but it seems quite reasonable.

I'd be happy to go with your suggestion and filter out lower headings and see if we get any pushback from other users (unlikely).

**Reply by JasonOui1** [reply]
Is there any chance this can be configurable? It looks like it was removed and I would like the H4 in the table of contents. It would be nice to have H5 and H6 support as well.

I think this could be a global instance/site setting but I do agree page level configuration would be great in the long term.

**Reply by almereyda** [reply]
Or per collection, per share ...

**Comment by fesja** [comment]
👌 thanks!
________________________________
De: Tom Moor ***@***.***>
Enviado: Monday, August 15, 2022 8:58:26 PM
Para: outline/outline ***@***.***>
Cc: Javier Escribano ***@***.***>; Author ***@***.***>
Asunto: Re: [outline/outline] Don't show all headings on the table of contents of each page (Discussion #3973)


I don't feel strongly about this but it seems quite reasonable.

I'd be happy to go with your suggestion and filter out lower headings and see if we get any pushback from other users (unlikely).
